src/Config/Settings.ts

```diff
@@ -49,6 +50,12 @@ export const TASK_FORMATS = {
 
 export type TASK_FORMATS = typeof TASK_FORMATS; // For convenience to make some typing easier
 
+// Time formats for the reminder settings.
+export const TIME_FORMATS = {
+    twelveHour: 'YYYY-MM-DD h:mm a',
+    twentyFourHour: 'YYYY-MM-DD HH:mm',
+};
```

Comment:
### Request: Only allow one format for the time - 24-hour clock

- Simplifies the Settings UI
- In particular, the current warning about not changing the format of the times is a bit of a smell for users
- Simplifies the task parsing/writing code
- Simplifies the Edit task modal
- Reduces the number of Jest tests required

Reply:
While it's easier, I think selecting time format to be pretty fundamental. I'm so accustomed to the 12hr format in my daily life that it would be kinda a hassle to be forced to use 24hr just for task reminders, and I'm sure the same thing could be said for the other camp. Ack that the current warning about changing formats isn't ideal, almost need some kinda of one type setup prompt when first installing to select your style and not allow flipping back and forth. Will think on this more

Reply:
That's a totally natural reaction.

I'll try and put it a different way.

I think the fastest way to get both formats supported is to make the first PR merge only support the simplest format and then to add the flexibility in a second PR.

I tend to find that the numbers of features in a PR have some kind of combinatorial cost - more features adds to the complexity of code, manual testing, automated tests, user documentation...

It's much easier to focus on one (less) thing at a time.

Hence this comment in the [Contributing Guide](https://publish.obsidian.md/tasks-contributing/Code/How+do+I+add+a+new+field+to+the+Task+class#Releasing%20a%20partial%20implementation):


> ## Releasing a partial implementation
> 
> For example, It's fine to have a first release of a feature without `sort by` and `group by`. In this case, add a feature request issue for the missing capabilities, and note in the documentation, for example:
> 
> "Sorting and grouping by blah is not yet supported. We are tracking this in [issue #nnnn](https://github.com/obsidian-tasks-group/obsidian-tasks/issues/nnn)."

Reply:
> ### Request: Only allow one format for the time - 24-hour clock


Ok, I just had a realisation that led to a big U-turn!

Having mulled over the test results here ...https://github.com/obsidian-tasks-group/obsidian-tasks/pull/1925/files#r1183336573 ... I can see ways of improving the handling of formatting, after this PR is merged.

So, let's stick with the user control of formats as you have it now, and just make the tests and documentation describe the current intended behaviour.